src/Interp.cc

```diff
@@ -2043,8 +2043,8 @@ static void do_defects(std::vector<Defect::Ptr> const & badList, // list of bad
 
 namespace {
     template<typename T>
-    struct Sort_ByX0 : public std::binary_function<typename T::Ptr const, typename T::Ptr const, bool> {
-        bool operator() (typename T::Ptr const a, typename T::Ptr const b) const {
+    struct Sort_ByX0 : public std::binary_function<std::shared_ptr<T> const, std::shared_ptr<T> const, bool> {
+        bool operator() (std::shared_ptr<T> const a, std::shared_ptr<T> const b) const {
```

Comment:
Maybe not, but converting shared pointers to unique pointers, raw pointers, and/or references is definitely outside the scope of the ticket. For one thing, it would require some more careful thought and testing than the simple syntax replacement I've done here.